src/module_2/groceries_analysis.md

```diff
@@ -0,0 +1,1811 @@
+     20  people_ex_baby                    float64
+     21  days_since_purchase_variant_id    float64
+     22  avg_days_to_buy_variant_id        float64
+     23  std_days_to_buy_variant_id        float64
+     24  days_since_purchase_product_type  float64
+     25  avg_days_to_buy_product_type      float64
+     26  std_days_to_buy_product_type      float64
+    dtypes: float64(19), int64(4), object(4)
+    memory usage: 593.4+ MB
+
+
+
+```python
+info_columns = ["variant_id", "order_id", "user_id", "created_at", "order_date"]
+target = "outcome"
+features_cols = [col for col in feature_frame.columns if col not in info_columns and col != target]
+categorical_cols = ['product_type', 'vendor']
+binary_cols = ['ordered_before','abandoned_before', 'active_snoozed', 'set_as_regular']
+numerical_cols = [col for col in features_cols if col not in categorical_cols and col not in binary_cols]
+
+print(f"Number of categorical variables : {len(categorical_cols)}")
+print(f"Number of binary variables : {len(binary_cols)}")
+print(f"Number of numerical variables : {len(numerical_cols)}")
+print(f"Number of info variables : {len(info_columns)}")
+
+```
+
+    Number of categorical variables : 2
+    Number of binary variables : 4
+    Number of numerical variables : 15
+    Number of info variables : 5
+
+
+
+```python
+feature_frame[target].value_counts()
+for col in binary_cols:
+    print(f"Value counts for {col}: {feature_frame[col].value_counts().to_dict()}")
+    print(f"Mean outcome by {col} value : {feature_frame.groupby(col)[target].mean().to_dict()}")
+    print("\n")
+```
+
+    Value counts for ordered_before: {0.0: 2819658, 1.0: 60891}
+    Mean outcome by ordered_before value : {0.0: 0.008223337723936732, 1.0: 0.1649669080816541}
+    
+    
+    Value counts for abandoned_before: {0.0: 2878794, 1.0: 1755}
+    Mean outcome by abandoned_before value : {0.0: 0.011106039542947498, 1.0: 0.717948717948718}
+    
+    
+    Value counts for active_snoozed: {0.0: 2873952, 1.0: 6597}
+    Mean outcome by active_snoozed value : {0.0: 0.011302554809544488, 1.0: 0.1135364559648325}
+    
+    
+    Value counts for set_as_regular: {0.0: 2870093, 1.0: 10456}
+    Mean outcome by set_as_regular value : {0.0: 0.010668992259135854, 1.0: 0.24971308339709258}
+    
+    
+
+
+
+```python
+# Correlation betweeen outcome and numerical features
+correlation = feature_frame[numerical_cols + [target]].corr()
+plt.figure(figsize=(20,10))
+sns.heatmap(correlation, annot=True, cmap='coolwarm')
+plt.title('Correlation Matrix')
+plt.show()
+```
+
+
+    
+![png](groceries_analysis_files/groceries_analysis_46_0.png)
+    
+
+
+
+```python
+# Distribution of numerical features by outcome
+rows = int(np.ceil(len(numerical_cols) / 3))
+fig, ax = plt.subplots(rows, 3, figsize=(20, 5 * rows))
+ax = ax.flatten()
+for i, col in enumerate(numerical_cols):
+    sns.kdeplot(feature_frame.loc[lambda x: x[target] == 0, col], label='0', ax=ax[i])
```

Comment:
The use of KDE for discrete variables is at least discouraging and at most misleading. For dsicrete variables bar plots is your way to go